### PR TITLE
add lab3

### PR DESCRIPTION
**Pull request by nipoks**

### CODE REVIEW

File: lab3.c [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        return NULL;
+    }
+
+    if (mkdir(data->dest, S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH) == -1) {
+        printf("Failed creat dest dir: %s\n", data->dest);
+        data_destructor(data);
+        return NULL;
+    }
+
+    pthread_t threadArr[1024]; //Ограниченное количество файлов в директории
```
Comment (k-stolyarov) [inline lab3.c]: If you have limited arrays please handle access to it's elements to avoid buffer overflows.

**Review by k-stolyarov — CHANGES_REQUESTED** [review]
Please share performance testing results on small/average/big structures and compare to "cp-r" command. You need to define what is small/moderate/average by yourself :)

File: lab3.c (hunk 1 above)
Comment (nipoks) [reply lab3.c]: В while добавлена проверка

**Review by k-stolyarov — APPROVED** [review]
Accepted

**Pull request merged**